ghost/admin/app/services/explore.js

```diff
@@ -90,26 +88,29 @@ export default class ExploreService extends Service {
         this.exploreWindowOpen = value;
     }
 
-    // Controls navigation to explore window modal which is triggered from the application UI.
-    // For example: pressing "View explore" link in navigation menu. It's main side effect is
-    // remembering the route from which the action has been triggered - "previousRoute" so it
-    // could be reused when closing the explore window
-    openExploreWindow(currentRoute, childRoute) {
+    openExploreWindow() {
         if (this.exploreWindowOpen) {
             // don't attempt to open again
             return;
         }
 
-        this.previousRoute = currentRoute;
+        // Begin loading the iframe and setting the src if it's not already set
+        this.ensureIframeIsLoaded();
 
         // Ensures correct "getIframeURL" calculation when syncing iframe location
         // in toggleExploreWindow
-        window.location.hash = childRoute || '/explore';
+        window.location.hash = '/explore';
 
-        this.router.transitionTo(childRoute || '/explore');
+        this.router.transitionTo('/explore');
         this.toggleExploreWindow(true);
     }
 
+    ensureIframeIsLoaded() {
+        if (this.getExploreIframe() && !this.getExploreIframe()?.src) {
+            this.getExploreIframe().src = this.getIframeURL();
+        }
+    }
+
     getExploreIframe() {
         return document.getElementById('explore-frame');
```

Comment:
@kevinansfield this works all ok (didn't implement any loading state yet), except for when Ember re-renders when in development mode. It appears the iframe is not available at the time of the re-render then and I'm unsure how to solve this without using the `did-insert` hook? Is there an example somewhere where we faced a similar issue? I couldn't find an equivalent in the glimmer docs to re-create the same livecyle hook 🤔

Reply:
I think it’s worth keeping the original `did-insert` behaviour to avoid too much change, my previous comment was more about adding another `did-insert` and the cleanup problems associated with that approach.

To get rid of all the `did-insert` usage I think would require a bit more of an architecture rethink. One thing that might be worth trying is switching `getIframeUrl()` to a `get iframeUrl()` and setting `src={{this.explore.iframeUrl}}` on the iframe itself so it's immediately loading with the right URL and doesn't need to be reset externally once rendered.

Reply:
Ah cool. I revived the `did-insert` hook again and converted `getIframeUrl` to a getter. Everything works fine and as expected to me now. Now I only need to implement a loading screen in Explore itself